### PR TITLE
SHOC warmT fix

### PR DESCRIPTION
**Pull request by bogensch**
This PR restores SHOC eddy diffusivity to their default formulations except when surface temperature appears to be running away.  This is triggered if surface temperature approaches a minimum allowable value and then extra mixing is temporarily applied.  The extra mixing is based on traditional stable PBL formulations that do not include stability correction terms.  

PR is a WIP because v1 is not compiling.  Also, b4b boiler plate has not been added.

### CODE REVIEW

File: components/eamxx/src/physics/shoc/tests/shoc_eddy_diffusivities_tests.cpp
Comment (tcclevenger) [inline components/eamxx/src/physics/shoc/tests/shoc_eddy_diffusivities_tests.cpp:28]: Property tests for `eddy_diffusivities` and `shoc_tke` need to be updated for new inputs.

File: components/eamxx/src/physics/shoc/tests/shoc_tke_tests.cpp
Comment (tcclevenger) [inline components/eamxx/src/physics/shoc/tests/shoc_tke_tests.cpp:26]: See comment in eddy_diffusivities test

**Review by tcclevenger — COMMENTED** [review]
@bogensch I left some comments. Basically
1. Can we just compute and store surface absolute temperature for each instead of computing every level?
2. Property tests will need to be updated to use the new input `tabs` (`eddy_diffusivities` and `shoc_tke`)
3. Does a property test need to be added for `compute_shoc_temperature`?

**Comment by bogensch** [comment]
Thanks @tcclevenger you are my v1 hero.  I will start to perform my v1 tests with this branch and will update the property tests.

File: components/eam/src/physics/cam/shoc.F90
Comment (bogensch) [reply components/eam/src/physics/cam/shoc.F90:3430]: Part of me wants to keep the whole 2d array over levels and columns in case I decide in the future that I want to access these values.  But I don't feel strongly about this so if you want to modify to only save the bottom value, that is fine with me.

File: components/eamxx/src/physics/shoc/tests/shoc_compute_shoc_temperature_tests.cpp
Comment (bogensch) [reply components/eamxx/src/physics/shoc/tests/shoc_compute_shoc_temperature_tests.cpp:28]: Probably.  This is something I will need to do.

**Comment by tcclevenger** [comment]
@bogensch No problem!

File: components/eam/src/physics/cam/shoc.F90
Comment (tcclevenger) [reply components/eam/src/physics/cam/shoc.F90:3430]: I'm ok leaving it for now then. If others object I can change it to surface values only.

**Comment by tcclevenger** [comment]
Some issues with floating point exceptions, should be solved now.

File: components/eamxx/src/physics/shoc/tests/shoc_eddy_diffusivities_tests.cpp
Comment (tcclevenger) [inline components/eamxx/src/physics/shoc/tests/shoc_eddy_diffusivities_tests.cpp:27]: Can you remove this TODO as well?

**Review by tcclevenger — COMMENTED** [review]
Thanks @bogensch. We'll need to find another reviewer since I'm not compromised having made commits.

**Comment by tcclevenger** [comment]
@bogensch There is an error in the new property check
```
-------------------------------------------------------------------------------
shoc_compute_shoc_temperature_property
-------------------------------------------------------------------------------
/home/tccleve/E3SM-Project/SCREAM/scream/components/eamxx/src/physics/shoc/tests/shoc_compute_shoc_temperature_tests.cpp:260
...............................................................................

/home/tccleve/E3SM-Project/SCREAM/scream/components/eamxx/src/physics/shoc/tests/shoc_compute_shoc_temperature_tests.cpp:124: FAILED:
  REQUIRE( SDS.tabs[offset] > SDS.thetal[offset] )
with expansion:
  290.0f > 290.0f
```
Is it ok to require `REQUIRE( SDS.tabs[offset] >= SDS.thetal[offset] )`

**Comment by bogensch** [comment]
thanks Conrad, yes I just fixed this.

**Review by tcclevenger — APPROVED** [review]
Changes look good. @AaronDonahue Someone else should probably also take a look since I contributed code myself.

**Comment by bogensch** [comment]
@tcclevenger, in addition before this PR gets merged I do want to verify that this code runs for four days at ne1024 without causing any problems.  My simulation is currently in queue and hopefully will go through tonight.  Will update when I am ready to sign off with this check.

**Comment by bogensch** [comment]
My ne1024 four day test completed without issue so I am now comfortable with this getting merged in.

**Review by crterai — APPROVED** [review]
I'm also okay to sign off on this from a simulation point of view, since the 4-day simulation ran without issue.

**Comment by crterai** [comment]
@tcclevenger - I think Peter's 4 day test was the last thing holding up this PR, so if it looks good to you, would you merge this PR to master?

**Comment by tcclevenger** [comment]
Thanks @crterai

**Comment by crterai** [comment]
Thanks @tcclevenger!

**Pull request merged**
